### ISSUE TITLE
Using Problem() on the basecontroller returns statuscode 200

### ISSUE THREAD

**Issue by eikc** [aspnet/AspNetCore#14663]
### Describe the bug
When returning a `BadRequest()` it will convert to a ProblemDetail and set the status code to 400. Whenever you use the `Problem()` the base controller will send a problem detail back but the status code is 200 OK but I would expect it to be a 500 as default. 

The same goes for ` return Problem("Details here", statusCode: StatusCodes.Status400BadRequest);` Here I would expect the returned status code to be 400 but it's actually returned as a 200 OK.

### To Reproduce
1. Using this version of ASP.NET Core '3.0.0'
2. Start a new asp.net web api solution and return a Problem from the controller or checkout repo: https://github.com/eikc/Aspnet-problem-detail
3. Run Application
4. The controller is returning 200 OK with a problem detail.

### Expected behavior
I Would expect the returned status code to match the status code of the problem detail and not a 200 OK.

### Screenshots
<img width="328" alt="problem-detail-bug" src="https://user-images.githubusercontent.com/5113271/66027660-aa80c200-e4fb-11e9-8fa7-dde6352b00cb.png">

### Additional context
```
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100
 Commit:    04339c3a26

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100\

Host (useful for support):
  Version: 3.0.0
  Commit:  7d57652f33

.NET Core SDKs installed:
  2.1.500 [C:\Program Files\dotnet\sdk]
  2.1.509 [C:\Program Files\dotnet\sdk]
  2.2.402 [C:\Program Files\dotnet\sdk]
  3.0.100 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by huysentruitw** [comment]
Ok if I pick up this issue?

**Comment by huysentruitw** [comment]
@pranavkm oops I see you self-assigned the issue. Please see my PR as a suggested solution. I still find it weird to have the StatusCode in the response header and as a property in the body...

**Comment by pranavkm** [comment]
This should be fixed as part of 3.1.0-preview2. Thanks @huysentruitw!
